src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,116 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(findByIdQuery)) {
+            statement.setLong(1, id);
+            generatedKeys = statement.executeQuery();
+            if (generatedKeys.next()) {
+                Book book = getBookFromResultSet(generatedKeys);
+                return Optional.of(book);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not find the book with id = " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String findAllQuery = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(findAllQuery)) {
+            generatedKeys = statement.executeQuery();
+            while (generatedKeys.next()) {
+                books.add(getBookFromResultSet(generatedKeys));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant create a connection to the DB", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String updateQuery = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(updateQuery)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not update the book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String deleteQuery = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(deleteQuery)) {
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not delete the book with id = " + id, e);
+        }
+    }
+
+    private Book getBookFromResultSet(ResultSet generatedKeys) {
+        try {
+            Book book = new Book();
+            book.setId(generatedKeys.getObject("id", Long.class));
+            book.setTitle(generatedKeys.getString("title"));
+            book.setPrice(generatedKeys.getBigDecimal("price"));
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not retrieve data from result set", e);
+        }
+    }

```

Comment:
You already have catch in methods where you call 'getBookFromResultSet' I think there is no reason to add another one